src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -2159,6 +2154,21 @@ private BoundExpression BindOutVariableArgument(DeclarationExpressionSyntax decl
                                         expressionVariableField, null, LookupResultKind.Viable, fieldType);
         }
 
+        /// <summary>
+        /// Returns true if a bad special by ref local was found.
+        /// </summary>
+        internal static bool CheckSpecialByRefLocal(Symbol containingSymbol, TypeSymbol type, DiagnosticBag diagnostics, SyntaxNode syntax)
```

Comment:
>CheckSpecialByRefLocal [](start = 29, length = 22)

The name feels confusing to me. I believe ByRef-ness of the local itself has nothing to do with the error.